### PR TITLE
[K8s] Decouple Test Suites

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
The PR aims to decouple the testing for k8s as we are currently deprecating hagrid.

Currently stack.test.integration.k8s tests the tests in 
1. notebooks/api/0.8
2. tests/integration/

This PR creates two tox tasks for split testing of k8s test suites
1. tox -e stack.test.notebook.k8s
2. tox -e stack.test.integration.k8s

## Affected Dependencies
K8s Testing

## How has this been tested?
Added new github action workflows for the same.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tests/integration/container_workload/pool_image_test.py [hunk 1]
```diff
@@ -103,18 +125,25 @@ def test_pool_launch(domain_1_port) -> None:
...
     docker_build_result = domain_client.api.services.worker_image.build(
         image_uid=worker_image.id,
         tag=docker_tag,
-        pull=False,
+        registry_uid=external_registry_uid,
     )
     assert isinstance(docker_build_result, SyftSuccess)
 
+    # Push Image to External registry
+    push_result = None
```
Comment (itstauq) [inline tests/integration/container_workload/pool_image_test.py]: Is this required?

File: tox.ini [hunk 2]
```diff
@@ -704,64 +701,116 @@ commands =
     sleep 30
 
     # wait for front end
```
Comment (itstauq) [inline tox.ini]: Shall we remove this too?

File: tox.ini [hunk 3]
```diff
@@ -704,64 +701,116 @@ commands =
...
+    devspace
+    kubectl
+    grep
+    sleep
+    bash
+    k3d
+    echo
+    tox
+setenv =
+    ORCHESTRA_DEPLOYMENT_TYPE = {env:ORCHESTRA_DEPLOYMENT_TYPE:k8s}
```
Comment (itstauq) [inline tox.ini:791]: TODO: change this to "remote" after merging #8788

**Review by itstauq — APPROVED** [review]
LGTM 🚀

File: tests/integration/container_workload/pool_image_test.py (hunk 1 above)
Comment (rasswanth-s) [reply tests/integration/container_workload/pool_image_test.py]: Not needed, addressed in [e32bd9b](https://github.com/OpenMined/PySyft/pull/8786/commits/e32bd9b5f07fdfd5553b1174a8f12aafbe6994cd)

File: tox.ini (hunk 2 above)
Comment (rasswanth-s) [reply tox.ini]: Done , addressed in [e32bd9b](https://github.com/OpenMined/PySyft/pull/8786/commits/e32bd9b5f07fdfd5553b1174a8f12aafbe6994cd)

**Review by itstauq — APPROVED** [review]

**Pull request merged**